wgpu-core/src/instance.rs

```diff
@@ -859,6 +871,10 @@ pub enum RequestDeviceError {
     TimestampNormalizerInitFailed(#[from] TimestampNormalizerInitError),
     #[error("Unsupported features were requested: {0:?}")]
     UnsupportedFeature(wgt::Features),
+    #[error(
+        "Experimental features were requested: {0:?}, but experimental features are not enabled"
```

Comment:
* How about "Some experimental features were requested" to clarify the distinction between 'an experimental feature' and 'experimental features' that's mostly implicit here?
* When using the `"problem: {data}"` style of error message, there shouldn't be any boilerplate text after the data. 

Two possible solutions:

```suggestion
        "Some experimental features, {0:?}, were requested, but experimental features are not enabled"
```

```suggestion
        "Some experimental features were requested, but experimental features are not enabled: {0:?}"
```

Reply:
I learned recently that the following is apparently possible (I think I dropped this from the final version of the change and am not sure I tested it, however):

```rust
#[error(
    "Shader requires capability {0:?}, which {is} unavailable either due to missing enable directives, or due to lack of platform support.",
    is = if .0.bits().is_power_of_two() { "is" } else { "are" },
)]
```

But then the other annoyance is the messages render as `requires capability Capabilities(FOO | BAR)`.

Reply:
Yeah I've pulled that off in a few places before, it's just rather verbose, so if we can get close enough, it's probably fine.

Reply:
@kpreid I applied the first one. I like keeping the actual features near the word experimental features.